home/models.py

```diff
@@ -1092,18 +1094,24 @@ class SVGToPNGMap(models.Model):
     png_image_file = models.ImageField(upload_to='svg-to-png-maps/')
 
     @classmethod
-    def get_png_image(cls, svg_path, fill_color, stroke_color=None):
+    def get_png_image(cls, svg_path, fill_color=None, stroke_color=None):
+        db_fill_color = fill_color or ''
+        db_stroke_color = stroke_color or ''
         try:
-            obj = cls.objects.get(svg_path=svg_path, fill_color=fill_color, stroke_color=stroke_color)
+            obj = cls.objects.get(svg_path=svg_path, fill_color=db_fill_color, stroke_color=db_stroke_color)
         except cls.DoesNotExist:
             try:
                 png_image = convert_svg_to_png_bytes(
                     svg_path, fill_color=fill_color, stroke_color=stroke_color, width=32)
             except:
                 logger.warning(f"Failed to convert SVG to PNG, file={svg_path}")
                 return None
-            obj = cls.objects.create(
-                svg_path=svg_path, fill_color=fill_color, stroke_color=stroke_color, png_image_file=png_image)
+            try:
+                obj = cls.objects.create(
+                    svg_path=svg_path, fill_color=db_fill_color,  stroke_color=db_stroke_color,
+                    png_image_file=png_image)
+            except IntegrityError:
```

Comment:
I think we ought to be super-paranoid here and catch any error, just in case. Please also log a warning message, so that we have some visibility on what's going on.

Reply:
done, logging exception message too.